apps/src/sites/studio/pages/blocks/edit.js

```diff
@@ -70,36 +72,65 @@ function initializeEditPage(defaultSprites) {
 
   if (blocks) {
     updateBlockPreview();
+    validateBlockConfig();
   }
+  setSubmitButtonState();
 
   $('.alert.alert-success')
     .delay(5000)
     .fadeOut(1000);
 }
 
 function onUpdateLinting(_, errors) {
-  const submitButton = document.querySelector('#block_submit');
   if (errors.length) {
-    submitButton.setAttribute('disabled', 'disabled');
+    hasLintingErrors = true;
   } else {
-    submitButton.removeAttribute('disabled');
+    hasLintingErrors = false;
   }
+  setSubmitButtonState();
 }
 
+const setSubmitButtonState = () => {
```

Comment:
I would move this function higher in the code, right before it's first usage for example. This way it'll be easier to read a code in the future. The function itself LGTM!

Reply:
Maybe a personal preference thing, but I generally like the style where the top of the file describes the high level functionality of the code (ie, in this case, what happens when the page loads, attach event handlers, etc.) And the bottom has more details of specific functions, if you need to see them. I can see where you're coming from though.